addons/fire/functions/fnc_burn.sqf

```diff
@@ -0,0 +1,352 @@
+    if (surfaceIsWater _unitPos && {(_unitPos#2) < 1}) then {
+        _intensity = 0;
+    };
+    
+    _fireParticle setDropInterval (0.01 max linearConversion [MAX_INTENSITY, MIN_INTENSITY, _intensity, 0.03, 0.1, false]);
+    _fireParticle setParticleParams [
+        ["\A3\data_f\ParticleEffects\Universal\Universal", 16, 10, 32], // sprite sheet values
+        "", // animation name
+        "Billboard", // particle type
+        1, // timer period
+        0.7, // lifetime
+        "destructionEffect2", // position
+        [0, 0, 1], // move velocity
+        0, // rotation velocity
+        10, // weight
+        7.9, // volume
+        1, // rubbing
+        [0.3, 0.3], // size
+        [
+            [1, 1, 1, -0],
+            [1, 1, 1, -1],
+            [1, 1, 1, -1],
+            [1, 1, 1, -1],
+            [1, 1, 1, -0]
+        ], // colour
+        [0.5, 1], // animation speed
+        1, // random dir period
+        0, // random dir intensity
+        "", // on timer script
+        "", // before destroy script
+        _unit, // particle source
+        0,
+        false,
+        0,
+        [[0.8, 0.6, 0.2, 1]] // emissive color
+    ];
+    _fireParticle setParticleRandom [
+        0.04 * _intensity, // life time
+        [0.05, 0.05, 2], // position
+        [0.05 * _intensity, 0.05 * _intensity, 0.05 * _intensity], // move velocity
+        0, // rotation velocity
+        0.06 * _intensity, // size
+        [0, 0, 0, 0], // color
+        0, // random direction period
+        0 // random direction intensity
+    ];
+    
+    _smokeParticle setParticleCircle [0, [0, 0, 0]];
+    _smokeParticle setParticleRandom [
+        0, // life time
+        [0.25, 0.25, 0], // position
+        [0.2, 0.2, 0], // move velocity
+        0, // rotation velocity
+        0.25, // size
+        [0, 0, 0, 0.1], // color
+        0, // random direction period
+        0 // random direction intensity
+    ];
+    _smokeParticle setParticleParams [
+        ["\A3\data_f\ParticleEffects\Universal\Universal", 16, 7, 48], // sprite sheet values
+        "", // animation name
+        "Billboard", // particle type
+        1, // timer period
+        8, // lifetime
+        [0, 0, 1.1], // position
+        [0, 0, 1], // move velocity
+        0, // rotation velocity
+        10, // weight
+        7.9, // volume
+        0.066, // rubbing
+        [1, 3, 6], // size
+        [
+            [0.5, 0.5, 0.5, 0.15],
+            [0.75, 0.75, 0.75, 0.075],
+            [1, 1, 1, 0]
+        ], // colour
+        [0.125], // animation speed
+        1, // random dir period
+        0, // random dir intensity
+        "", // on timer script
+        "", // before destroy script
+        _unit // particle source
+    ];
+    _smokeParticle setDropInterval 0.15;
+    
+    _fireLight setLightBrightness ((_intensity * 3) / 10);
+    _lightFlare setLightBrightness (_intensity / 30);
+    
+    private _distanceToUnit = (_unit distance ACE_PLAYER);
```

Comment:
Why is `ACE_PLAYER` in all caps? :thinking: 
Unless I missed a define, this should be `ace_player`.

Reply:
At some point I must of seen some sample code that used `ACE_PLAYER` and I've used it ever since. I am pretty sure that the all caps version is scattered across my code